### PR TITLE
feat: Create infra as part of the streaming ingestion

### PR DESCRIPTION
**Pull request by EXPEbdodla**
# What this PR does / why we need it:
1. Creates the infra as part of spark kafka processor streaming ingestion

# Which issue(s) this PR fixes:

# Misc

### CODE REVIEW

File: sdk/python/feast/infra/contrib/spark_kafka_processor.py [hunk 1]
```diff
@@ -116,9 +117,27 @@ def __init__(
         # data_source type has been checked to be an instance of KafkaSource.
         self.data_source: KafkaSource = self.data_source  # type: ignore
 
+    def _create_infra_if_necessary(self):
```
Comment (vanitabhagwat) [inline sdk/python/feast/infra/contrib/spark_kafka_processor.py:120]: Would it be better to name it something like setup_infra_if_lazy_creation?

File: sdk/python/feast/infra/contrib/spark_kafka_processor.py [hunk 2]
```diff
@@ -116,9 +117,27 @@ def __init__(
...
+        if self.fs.config.online_store is not None and getattr(
+            self.fs.config.online_store, "lazy_table_creation", False
+        ):
+            print(
+                f"Online store {self.fs.config.online_store.__class__.__name__} supports lazy table creation and it is enabled"
+            )
```
Comment (zabarn) [inline sdk/python/feast/infra/contrib/spark_kafka_processor.py:126]: Should we use print or logger?

**Review by zabarn — APPROVED** [review]

File: sdk/python/feast/infra/contrib/spark_kafka_processor.py (hunk 2 above)
Comment (EXPEbdodla) [reply sdk/python/feast/infra/contrib/spark_kafka_processor.py:126]: Currently print statement inside the spark tasks. Loggers need to be configured but it's not the case currently.

File: sdk/python/feast/infra/contrib/spark_kafka_processor.py (hunk 1 above)
Comment (EXPEbdodla) [reply sdk/python/feast/infra/contrib/spark_kafka_processor.py:120]: Leaving it for now. We are looking for better ways to manage infra. This is not a sustainable solution.

**Pull request merged**